apps/mobile/src/queries/transaction/use-check-utxos.ts

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useState } from 'react';
+
+import { TransactionInput } from '@scure/btc-signer/psbt';
+import { bytesToHex } from '@stacks/common';
+
+import { BitcoinClient, getNumberOfInscriptionOnUtxoUsingOrdinalsCom } from '@leather.io/query';
+import { isUndefined } from '@leather.io/utils';
+
+import { useBitcoinClient } from '../clients/bitcoin-client';
+import { useCurrentNetworkState, useIsLeatherTestingEnv } from '../leather-query-provider';
+
+class PreventTransactionError extends Error {
+  constructor(message: string) {
+    super(message);
+    // eslint-disable-next-line lingui/no-unlocalized-strings
+    this.name = 'PreventTransactionError';
+  }
+}
+interface CheckInscribedUtxosByBestinslotArgs {
+  inputs: TransactionInput[];
+  txids: string[];
+  client: BitcoinClient;
+}
+
+async function checkInscribedUtxosByBestinslot({
+  inputs,
+  txids,
+  client,
+}: CheckInscribedUtxosByBestinslotArgs): Promise<boolean> {
+  /**
+   * Get the list of inscriptions moving on a transaction
+   * @see https://docs.bestinslot.xyz/reference/api-reference/ordinals-and-brc-20-and-bitmap-v3-api-mainnet+testnet/inscriptions
+   */
+  const inscriptionIdsList = await Promise.all(
+    txids.map(id => client.BestInSlotApi.getInscriptionsByTransactionId(id))
+  );
+
+  const inscriptionIds = inscriptionIdsList.flatMap(inscription =>
+    inscription.data.map(data => data.inscription_id)
+  );
+
+  const inscriptionsList = await Promise.all(
+    inscriptionIds.map(id => client.BestInSlotApi.getInscriptionById(id))
+  );
+
+  const hasInscribedUtxos = inscriptionsList.some(resp => {
+    return inputs.some(input => {
+      if (!input.txid) throw new Error('Transaction ID is missing in the input');
+      const idWithIndex = `${bytesToHex(input.txid)}:${input.index}`;
+      return resp.data.satpoint.includes(idWithIndex);
+    });
+  });
+
+  return hasInscribedUtxos;
+}
+
+export function useCheckUnspendableUtxos(blockTxAction?: () => void) {
+  const client = useBitcoinClient();
+  const [isLoading, setIsLoading] = useState(false);
+  const { isTestnet } = useCurrentNetworkState();
+  const isTestEnv = useIsLeatherTestingEnv();
+
+  const preventTransaction = useCallback(() => {
+    if (blockTxAction) return blockTxAction();
+    throw new PreventTransactionError(
```

Comment:
You will need to add a translation key for this error message so you should just use `BitcoinError` and add a translation to `formatBitcoinError`